app/controllers/api/status_controller.rb

```diff
@@ -3,11 +3,32 @@ class Api::StatusController < Api::ApplicationController
 
   def check
     if params[:projects].any?
+      # Try to get all the projects passed in.
       @projects = params[:projects].group_by{|project| project[:platform] }.map do |platform, projects|
         projects.each_slice(1000).map do |slice|
           Project.platform(platform).where(name: slice.map{|project| project[:name] }.uniq).includes(:repository, :versions)
         end.flatten.compact
       end.flatten.compact
+
+      # If there are any that don't get found, downcase them and try again
+      if params[:projects].length != @projects.length
+        # "slugify" the platform projects that we found
```

Comment:
I'm a little confused as to why you need to do this... is it just to reduce what you're iterating over to be a little simpler? If so, you could just have a list of `[platform, name]` pairs as `list.include?` works with finding a list inside a list in ruby